proposal.md

```diff
@@ -0,0 +1,283 @@
+# Reflect Library Improvements Proposal
+
+## Background
+
+In Eventide's libraries, Reflect has 6 use cases, but 4 of them do not utilize it in the designed way. Instead of making a use of the reflection's actuator, the constant is pulled from the reflection.
+
+The actuator is designed to pass the subject into the method being invoked. Subject could be overriden when needed.
+
+```ruby
+class SomeThing
+  module Validate
+    def self.call(some_thing)
+      # validate some_thing
+    end
+  end
+end
+
+some_thing = SomeThing.new
+
+validator_reflection = Reflect.(some_thing, :Validate)
+
+validator_reflection.(:call)
+```
+
+However, the actuator doesn't support methods with zero arity or methods with arity greater than one. Due to these limitations, the constant is extracted from the reflection, bypassing the actuator.
```

Comment:
Consider: "Due to this limitation, the constant is often pulled from the reflection, ..."